docs/machine-learning/tutorials/object-detection-custom-vision-onnx.md

```diff
@@ -0,0 +1,388 @@
+    ```
+
+In the following code blocks, we will be adding within the `foreach` loop.
+
+1. First, we will generate the predicted image name based on the name of the test image.
+
+    ```csharp
+    var predictedImage = $"{Path.GetFileName(image)}-predicted.jpg";
+    ```
+
+1. Next, we will create a `FileStream` of our image and convert it to a `Bitmap`.
+
+    ```csharp
+    using (var stream = new FileStream(image, FileMode.Open))
+    {
+        testImage = (Bitmap)Image.FromStream(stream);
+    }
+    ```
+
+1. Now we can call the `Predict` method on the prediction engine.
+
+    ```csharp
+    var prediction = predictionEngine.Predict(new StopSignInput { Image = testImage });
+    ```
+
+1. With our prediction, we can get the bounding boxes. We will use the `Chunk` LINQ method to determine how many objects the model has detected. We do this by taking the count of the predicted bounding boxes and divide that by the number of labels that were predicted. For example, if we had three objects detected in an image, there would be 12 items in the `BoundingBoxes` array and three labels predicted. The `Chunk` method would then give us three arrays of four to represent the bounding boxes for each object.
+
+    ```csharp
+    var boundingBoxes = prediction.BoundingBoxes.Chunk(prediction.BoundingBoxes.Count() / prediction.PredictedLabels.Count());
+    ```
+
+1. Next, we will capture the original width and height of the images used for prediction.
+
+    ```csharp
+    var originalWidth = testImage.Width;
+    var originalHeight = testImage.Height;
+    ```
+
+1. With the bounding boxes chunk, we can loop through those to calculate where in the image to draw the boxes. For that, create a `for` loop based on the count of the bounding boxes chunk.
+
+    ```csharp
+    for (int i = 0; i < boundingBoxes.Count(); i++)
+    {
+    }
+    ```
+
+1. Within the `for` loop we will perform some calculations to get the position of the x, y, width, and height of the box for us to draw on the image. The first thing we need to do is to get the set of bounding boxes. We can do this by using the `ElementAt` method.
+
+    ```csharp
+    var boundingBox = boundingBoxes.ElementAt(i);
+    ```
+
+1. With the current bounding box, we can now calculate where to draw the box. We will use the original image width for the first and third elements of the bounding box, and the original image height for the second and fourth elements.
+
+    ```csharp
+    var left = boundingBox[0] * originalWidth;
+    var top = boundingBox[1] * originalHeight;
+    var right = boundingBox[2] * originalWidth;
+    var bottom = boundingBox[3] * originalHeight;
+    ```
+
+1. Now we can calculate the width and the height of the box to draw around the detected object within the image. The x and y items will be the `left` and `top` variables from the previous calculation. We will use the `Math.Abs` method to get the absolute value from the width and height calculations in case it is negative.
```

Comment:
```suggestion
1. Calculate the width and the height of the box to draw around the detected object within the image. The x and y items will be the `left` and `top` variables from the previous calculation. Use the `Math.Abs` method to get the absolute value from the width and height calculations in case it is negative.
```